### PR TITLE
chore(storybook): move storybook setup to TypeScript

### PR DESCRIPTION
**Pull request by jcfranco**
**Related Issue:** #687 

## Summary

This moves our Storybook setup to TypeScript.

Also:

* adds `.npmrc` to standardize dependency installs
* fixes sorting of stories
* splits up `tsconfig.json` with common base

Note: `*.stories.js` files will no longer get picked up, but this can be fixed by simply renaming the file to `*.stories.ts`. 🥇

### CODE REVIEW

File: .storybook/preview.ts [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+import centered from "@storybook/addon-centered/html";
+import theme from "./theme";
+import { titlelessDocsPage } from "./utils";
+
+addDecorator(centered);
+addParameters({
+  backgrounds: [{ name: "Light", value: "#f8f8f8", default: true }],
+  options: {
+    theme,
+    storySort: (a, b) => (a[1].kind === b[1].kind ? 0 : a[1].id.localeCompare(b[1].id, undefined, { numeric: true }))
```
Comment (paulcpederson) [inline .storybook/preview.ts:11]: how does this work?

Comment (driskull) [reply .storybook/preview.ts:11]: That sorts the stories by type and then numerically/alphabetically

Comment (jcfranco) [reply .storybook/preview.ts:11]: I got it from the [Storybook doc](https://storybook.js.org/docs/configurations/options-parameter/#sorting-stories). I tested, but didn't look into it. However, after digging a little bit this morning, I [found](https://github.com/storybookjs/storybook/pull/9188/files) a simpler way:

```ts
storySort: {
  method: "alphabetical"
}
```

Comment (driskull) [reply .storybook/preview.ts]: @jcfranco we actually have to use this to get the storybook sorted with mdx files at top, followed by components and then app components.

```ts
storySort: (a, b) => {
      const sectionA = a[1].id.split("-")[0];
      const sectionB = b[1].id.split("-")[0];

      return sectionB.localeCompare(sectionA);
    },
```

Comment (jcfranco) [reply .storybook/preview.ts]: Got it. I'll tackle that in the port PR.

**Pull request merged**